prediction_market_agent/agents/microchain_agent/microchain_agent.py

```diff
@@ -9,39 +9,60 @@
 )
 from prediction_market_agent.utils import APIKeys
 
+SYSTEM_PROMPT = """Act as a agent to maximise your profit.
 
-def main(
+You can use the following functions:
+
+{engine_help}
+
+Only output valid Python function calls.
+"""
+
+
+def get_agent(
+    market_type: MarketType,
+    model: str,
     api_base: str = "https://api.openai.com/v1",
-    model: str = "gpt-4-turbo-preview",
-) -> None:
+) -> Agent:
     engine = Engine()
     engine.register(Reasoning())
     engine.register(Stop())
     for function in MISC_FUNCTIONS:
         engine.register(function())
     for function in MARKET_FUNCTIONS:
-        engine.register(function(market_type=MarketType.OMEN))
+        engine.register(function(market_type=market_type))
     for function in OMEN_FUNCTIONS:
         engine.register(function())
-
     generator = OpenAIChatGenerator(
         model=model,
         api_key=APIKeys().openai_api_key.get_secret_value(),
         api_base=api_base,
         temperature=0.7,
     )
     agent = Agent(llm=LLM(generator=generator), engine=engine)
-    agent.prompt = f"""Act as a agent to maximise your profit. You can use the following functions:
-    
-    {engine.help}
-    
-    
-    Only output valid Python function calls.
-    
-    """
-
-    agent.bootstrap = ['Reasoning("I need to reason step-by-step")']
-    agent.run(iterations=10)
+    agent.prompt = SYSTEM_PROMPT.format(engine_help=engine.help)
+    agent.bootstrap = [
+        'Reasoning("I need to reason step by step. Start by assessing my '
+        'current position and balance.")'
+    ]
```

Comment:
Should we also add some explanation about how to make money in such a market?
Maybe that's the reason why it just bets everything in a simple market, it has no idea what it's doing

Reply:
Yeah maybe. Need to have more of a play around with it to see what performs best. I don't think we need to be too worried about this for now. Btw I added a failsafe `MAX_AMOUNT` to `BuyTokens` so it can't do that now :D

Reply:
works for me